tests/v2/validation/networking/connectivity/network_policy_test.go

```diff
@@ -0,0 +1,145 @@
+	"github.com/stretchr/testify/suite"
+)
+
+type NetworkPolicyTestSuite struct {
+	suite.Suite
+	session     *session.Session
+	client      *rancher.Client
+	project     *management.Project
+	clusterName string
+	namespace   *v1.SteveAPIObject
+	steveClient *v1.Client
+}
+
+const (
+	nodeRole = "control-plane"
+	// Ping once
+	pingCmd        = "ping -c 1"
+	successfulPing = "0% packet loss"
+)
+
+func (n *NetworkPolicyTestSuite) TearDownSuite() {
+	n.session.Cleanup()
+}
+
+func (n *NetworkPolicyTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	n.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(n.T(), err)
+
+	n.client = client
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmpty(n.T(), clusterName, "Cluster name to install is not set")
+	n.clusterName = clusterName
+
+	cluster, err := clusters.NewClusterMeta(client, clusterName)
+	require.NoError(n.T(), err)
+
+	projectConfig := &management.Project{
+		ClusterID: cluster.ID,
+		Name:      pingPodProjectName,
+	}
+
+	createdProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(n.T(), err)
+	require.Equal(n.T(), createdProject.Name, pingPodProjectName)
+	n.project = createdProject
+
+	names := newNames()
+
+	namespaceName := names.random["namespaceName"]
+	daemonsetName := names.random["daemonsetName"]
+
+	n.T().Logf("Creating namespace with name [%v]", namespaceName)
+	n.namespace, err = namespaces.CreateNamespace(n.client, namespaceName, "{}", map[string]string{}, map[string]string{}, n.project)
+	require.NoError(n.T(), err)
+	assert.Equal(n.T(), n.namespace.Name, namespaceName)
+
+	n.steveClient, err = n.client.Steve.ProxyDownstream(n.project.ClusterID)
+	require.NoError(n.T(), err)
+
+	testContainerPodTemplate := newPodTemplateWithTestContainer()
+
+	n.T().Logf("Creating a daemonset with the test container with name [%v]", daemonsetName)
+	daemonsetTemplate := shepworkloads.NewDaemonSetTemplate(daemonsetName, n.namespace.Name, testContainerPodTemplate, true, nil)
+	createdDaemonSet, err := n.steveClient.SteveType(workloads.DaemonsetSteveType).Create(daemonsetTemplate)
+	require.NoError(n.T(), err)
+	assert.Equal(n.T(), createdDaemonSet.Name, daemonsetName)
+
+	n.T().Logf("Waiting for daemonset [%v] to have expected number of available replicas", daemonsetName)
+	err = charts.WatchAndWaitDaemonSets(n.client, n.project.ClusterID, n.namespace.Name, metav1.ListOptions{})
+	require.NoError(n.T(), err)
+}
+
+func (n *NetworkPolicyTestSuite) TestPingPodsFromCPNode() {
+
+	// Get downstream cluster wrangler context
+	wc, err := n.client.WranglerContext.DownStreamClusterWranglerContext(n.project.ClusterID)
+	require.NoError(n.T(), err)
+
+	pods, err := wc.Core.Pod().List(n.namespace.Name, metav1.ListOptions{})
+	assert.NoError(n.T(), err)
+	assert.NotEmpty(n.T(), pods)
+
+	_, stevecluster, err := clusters.GetProvisioningClusterByName(n.client, n.clusterName, provisioninginput.Namespace)
```

Comment:
Since we're already using the global struct c.clusterName, what do you think about updating it to c.cluster instead? so that we can avoid fetching the cluster in each test and it would be easier to manage if we add more tests in the future.

Reply:
Can you give me an example of this?

Reply:
something like this: https://github.com/rancher/rancher/blob/main/tests/v2/validation/rbac/rbac_test.go#L26 and then in the setup suite adding this https://github.com/rancher/rancher/blob/main/tests/v2/validation/rbac/rbac_test.go#L47... c.cluster will have the cluster ID as well.